modules/ting_das/includes/ting_das.client.inc

```diff
@@ -0,0 +1,430 @@
+   * Set date well id.
+   *
+   * @param string $pid
+   *   The date well id for the material requested.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setPid(string $pid) {
+    $this->pid = $pid;
+    return $this;
+  }
+
+  /**
+   * The data well id.
+   *
+   * @return string|null
+   *   The data well pid if set else NULL.
+   */
+  public function getPid() {
+    return $this->pid;
+  }
+
+  /**
+   * Set the users loaner id.
+   *
+   * @param string $userLoanerId
+   *   The users loaner id (borrow id/CPR).
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setUserLoanerId(string $userLoanerId) {
+    $this->userLoanerId = $userLoanerId;
+    return $this;
+  }
+
+  /**
+   * Get the user loaner id.
+   *
+   * @return string|null
+   *   The loaner id or null if not set.
+   */
+  public function getUserLoanerId() {
+    return $this->userLoanerId;
+  }
+
+  /**
+   * Set the username (full name of the user).
+   *
+   * @param string $userName
+   *   The users full name.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setUserName(string $userName) {
+    $this->userName = $userName;
+    return $this;
+  }
+
+  /**
+   * Get the users full name.
+   *
+   * @return string|null
+   *   The users full name.
+   */
+  public function getUserName() {
+    return $this->userName;
+  }
+
+  /**
+   * Set the users e-mail-address.
+   *
+   * @param string $mail
+   *   The users e-mail-address.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setUserMail(string $mail) {
+    $this->mail = $mail;
+    return $this;
+  }
+
+  /**
+   * Get the users e-mail-address.
+   *
+   * @return string|null
+   *   The users e-mail-address.
+   */
+  public function getUserMail() {
+    return $this->mail;
+  }
+
+  /**
+   * The interest date for the user.
+   *
+   * The date until the user is not longer interested in getting the material.
+   *
+   * @param string $userInterestDays
+   *   The users interest date in number of days.
+   *
+   * @return TingDasCopyRequest
+   *   The current request object.
+   */
+  public function setUserInterestDate(string $userInterestDays) {
+    $this->userInterestDays = $userInterestDays;
+    return $this;
+  }
+
+  /**
+   * Get the users interest date.
+   *
+   * @param string $format
+   *   The date format to use.
+   *
+   * @return string|null
+   *   The users interest date in the format given from now.
+   */
+  public function getUserInterestDate($format = 'Y-m-d') {
+    $date = strtotime('+' . $this->userInterestDays . ' day');
+    return date($format, $date);
+  }
+
+  /**
+   * Set the agency id for pickup.
+   *
+   * @param string $pickupAgencyId
+   *   The pickup agency id.
+   *
+   * @Note: Not sure why this exists as all articles are sent by mail to the
+   *        end user.
```

Comment:
If we do not use it then I think it should be removed along with the getter. If it is needed when sending requests we can keep it internally.

Reply:
As the service is not documented I did not known at the time why. It turns out that do to copyright issues an article may be delivered to an location (pickup agency) as a paper copy.

So comment has been removed.